pkg/standalone-utils/test/VaultActions.test.ts

```diff
@@ -1161,4 +1190,324 @@ describe('VaultActions', function () {
+          sender = relayer;
+          await tokens.DAI.transfer(relayer, amountDAI, { from: user });
+          await tokens.SNX.transfer(relayer, amountSNX, { from: user });
+          await approveVaultForRelayer(relayerLibrary, user, tokens);
+        });
+
+        itTestsUserBalance();
+      });
+
+      function itTestsUserBalance() {
+        it('sends immediate amounts', async () => {
+          // Internal balance of sender doesn't change
+          // Tokens are transferred from sender to recipient's internal balance
+
+          await expectBalanceChange(
+            () =>
+              relayer.connect(user).multicall([
+                encodeManageUserBalance({
+                  ops: [
+                    { kind: UserBalanceOpKind.DepositInternal, asset: tokens.DAI.address, amount: amountDAI, sender },
+                    { kind: UserBalanceOpKind.DepositInternal, asset: tokens.SNX.address, amount: amountSNX, sender },
+                  ],
+                }),
+              ]),
+            tokens,
+            [
+              {
+                account: TypesConverter.toAddress(sender),
+                changes: {
+                  DAI: 0,
+                  SNX: 0,
+                },
+              },
+              {
+                account: TypesConverter.toAddress(recipient),
+                changes: {
+                  DAI: amountDAI,
+                  SNX: amountSNX,
+                },
+              },
+            ],
+            vault.instance
+          );
+        });
+
+        it('stores vault deltas as chained references', async () => {
+          const receipt = await (
+            await relayer.connect(user).multicall([
+              encodeManageUserBalance({
+                ops: [
+                  { kind: UserBalanceOpKind.DepositInternal, asset: tokens.DAI.address, amount: amountDAI, sender },
+                  { kind: UserBalanceOpKind.DepositInternal, asset: tokens.SNX.address, amount: amountSNX, sender },
+                ],
+                outputReferences: [
+                  { index: 0, key: toChainedReference(0) },
+                  { index: 1, key: toChainedReference(1) },
+                ],
+              }),
+            ])
+          ).wait();
+
+          expectEvent.inIndirectReceipt(receipt, vault.instance.interface, 'InternalBalanceChanged', {
+            user: TypesConverter.toAddress(recipient),
+            token: tokens.DAI.address,
+            delta: amountDAI,
+          });
+
+          expectEvent.inIndirectReceipt(receipt, vault.instance.interface, 'InternalBalanceChanged', {
+            user: TypesConverter.toAddress(recipient),
+            token: tokens.SNX.address,
+            delta: amountSNX,
+          });
+
+          await expectChainedReferenceContents(relayer, toChainedReference(0), amountDAI);
+
+          await expectChainedReferenceContents(relayer, toChainedReference(1), amountSNX);
+        });
```

Comment:
This can probably be split in two tests:
- one that checks the events in the receipt
- one that checks the output references